packages/core/src/tools/utils.spec.ts

```diff
@@ -317,3 +327,189 @@ describe('findCommaSeparatedValue', () => {
+      const source = [1]
+      expect(mergeInto({}, source)).toEqual(source)
+    })
+
+    it('should mutate and return destination if it is an array', () => {
+      const destination = ['destination']
+      const source = ['source']
+      const result = mergeInto(destination, source)
+      expect(result).toBe(destination)
+      expect(result).toEqual(source)
+    })
+  })
+
+  describe('source is an object', () => {
+    it('should create a new object if destination is undefined', () => {
+      const source = {}
+      const result = mergeInto(undefined, source)
+      expect(result).not.toBe(source)
+      expect(result).toEqual(source)
+    })
+
+    it('should return the copy of source if the destination is not an object', () => {
+      const source = { a: 1 }
+      expect(mergeInto([], source)).toEqual(source)
+    })
+
+    it('should mutate and return destination if it is an object', () => {
+      const destination = {}
+      const source = { a: 'b' }
+      const result = mergeInto(destination, source)
+      expect(result).toBe(destination as any)
+      expect(result).toEqual(source)
+    })
+  })
+})
+
+describe('combine', () => {
+  it('should deeply add and replace keys', () => {
+    const sourceA = { a: { b: 'toBeReplaced', c: 'source a' } }
+    const sourceB = { a: { b: 'replaced', d: 'source b' } }
+    expect(combine(sourceA, sourceB)).toEqual({ a: { b: 'replaced', c: 'source a', d: 'source b' } })
+  })
+
+  it('should not replace with undefined', () => {
+    expect(combine({ a: 1 }, { a: undefined as number | undefined })).toEqual({ a: 1 })
+  })
+
+  it('should replace a sub-value with null', () => {
+    expect(combine({ a: {} }, { a: null as any })).toEqual({ a: null })
+  })
+
+  it('should ignore null arguments', () => {
+    expect(combine({ a: 1 }, null)).toEqual({ a: 1 })
+  })
+
+  it('should merge arrays', () => {
+    const sourceA = [{ a: 'source a' }, 'extraString'] as any
+    const sourceB = [{ b: 'source b' }] as any
+    expect(combine(sourceA, sourceB)).toEqual([{ a: 'source a', b: 'source b' }, 'extraString'])
+  })
+
+  it('should merge multiple objects', () => {
+    expect(combine({ a: 1 }, { b: 2 }, { c: 3 })).toEqual({ a: 1, b: 2, c: 3 })
+  })
+
+  it('should not keep references on objects', () => {
+    const source = { a: { b: 1 } }
+    const result = combine({}, source)
+    expect(result.a).not.toBe(source.a)
+  })
+
+  it('should not keep references on arrays', () => {
+    const source = { a: [1] }
+    const result = combine({}, source)
+    expect(result.a).not.toBe(source.a)
+  })
+})
+
+describe('deepClone', () => {
+  it('pass-through non-objects', () => {
+    expect(deepClone('test')).toBe('test')
+    expect(deepClone(true)).toBe(true)
+    expect(deepClone(null)).toBe(null)
+    const fn = () => null
+    expect(deepClone(fn)).toBe(fn)
+    expect(deepClone(1)).toBe(1)
+    expect(deepClone(undefined)).toBe(undefined)
+  })
+
+  it('clones array', () => {
```

Comment:
Could we keep some explanation of the behavior in the test description?
`clones array` VS `should return different references for arrays items`